### ISSUE TITLE
Wrong indentation if line contains `'[`

### ISSUE THREAD

**Issue by m1dnight** [cmhughes/latexindent.pl#178]
I'm using Mathpar in my code, and latexindent aligns it properly. However,  I think that when there is a sequence with `'` and `[` in the text, it does not insert  tab before that line.

This example: 

```
\begin{mathpar}
	\inferrule[E-Realize]
	{
		this will align
	}
	{
		foobar
	}

	\inferrule[E-StreamGet]
	{
		this will align
	}
	{
		[x] contains [ and ]
	}

	\inferrule[E-StreamGet]
	{
		this will align too
	}
	{
		s[]
	}

	\inferrule[E-StreamGet]
	{
	this does not!
	}
	{
	'[
	}

\end{mathpar}
```

**Comment by cmhughes** [comment]
Many thanks for this, it looks interesting. I'll take a look over the next
few days and get back to you.

**Comment by cmhughes** [comment]
Possibly we might use specialBeginEnd with verbatim:

https://latexindentpl.readthedocs.io/en/latest/sec-default-user-local.html#lst-special-verb1-yaml

I'll explore further later on, feel free to do the same :)

**Comment by cmhughes** [comment]
Starting with

`myfile.tex`
-
```
\inferrule[E-StreamGet]
	{
	this does not!
	}
	{
	'[
	}
```

If you use the following `YAML`

`m1dnight.yaml`
-
```
specialBeginEnd:
    dashBrace:
        begin: \'
        end: \[
        lookForThis: verbatim
```
and call
```
     latexindent.pl myfile.tex m1dnight.yaml
```
then you receive
```
\inferrule[E-StreamGet]
{
	this does not!
}
{
	'[
}
```

Why does this work?
-
Referencing https://latexindentpl.readthedocs.io/en/latest/sec-default-user-local.html#lst-special-verb1-yaml we have specified a `specialBeginEnd` block that starts with `'` and finishes with `[`, and furthermore we have specified that it should be treated as a `verbatim` block. This means that it will be stored before any of the indentation operations, and then replaced at the end after the indentation has been performed. 

Without this specification, the arguments regular expression (see https://github.com/cmhughes/latexindent.pl/blob/master/LatexIndent/Arguments.pm) will fail, because the `'[` is an unmatched brace. 

Hopefully this is as you would like :)
